packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -249,6 +249,41 @@ class RenderEditable extends RenderBox {
 
   Rect _lastCaretRect;
 
+  /// Track the visbility of the start of the selected text.
+  ///
+  /// For example, if the text contains "Hello World", and the user selects
+  /// "Hello", then scrolls so only "World" is visible, this will become
+  /// 'false'. If the user scrolls back so that the "H" is visible again, this
+  /// will become 'true'.
+  final ValueNotifier<bool> selectionStartInView = ValueNotifier<bool>(true);
+
+  /// Track the visbility of the end of the selected text.
+  ///
+  /// For example, if the text contains "Hello World", and the user selects
+  /// "World", then scrolls so only "Hello" is visible, this will become
+  /// 'false'. If the user scrolls back so that the "d" is visible again, this
+  /// will become 'true'.
+  final ValueNotifier<bool> selectionEndInView = ValueNotifier<bool>(true);
+
+  void _updateSelectionExtentsVisibility(Offset effectiveOffset) {
+    final Rect visibleRegion = Offset.zero & size;
+
+    final bool startExtentVisible = visibleRegion.contains(
+        _textPainter.getOffsetForCaret(
+            TextPosition(offset: _selection.start, affinity: _selection.affinity),
+            Rect.zero) +
+        effectiveOffset);
+
+    final bool endExtentVisible = visibleRegion.contains(
+        _textPainter.getOffsetForCaret(
+            TextPosition(offset: _selection.end, affinity: _selection.affinity),
+            Rect.zero) +
```

Comment:
nit: flutter style; see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#if-you-have-a-newline-after-some-opening-punctuation-match-it-on-the-closing-punctuation

Reply:
Done.